### PR TITLE
[Libtomcrypt, libtommath] Add new port

### PR DESCRIPTION
**Pull request by FredeJ**
I added libtommath and libtomcrypt.

I tried updating the solution file  to be a newer version so it could be build with the `vcpkg_build_msbuild` command. However I was advised to use the existing `makefile.msvc` instead using `nmake` ( see https://github.com/libtom/libtomcrypt/pull/517 ).

I made it build by inelegantly including `${CURRENT_PACKAGES_DIR}/..` - what is the solution for elegantly including the currently built libraries?

I ran into some issues with the approach, namely building a debug version (still not sure I'm doing that right) and building for x64 (unsolved as of yet). Additionally, I'm not completely sure how to go about building for linux and filtering correctly on the triplets. 

Instead of just giving up and concluding that it's too much work to get non-cmake/msbuild packages to work with vcpkg I figured it would upload what I have now to get some feedback on how to proceed.

### CODE REVIEW

File: ports/libtomcrypt/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,55 @@
+include(vcpkg_common_functions)
```
Comment (LilyWangL) [inline ports/libtomcrypt/portfile.cmake]: include(vcpkg_common_functions) is deprecated. Please remove it.

File: ports/libtommath/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,52 @@
+include(vcpkg_common_functions)
```
Comment (LilyWangL) [inline ports/libtommath/portfile.cmake]: include(vcpkg_common_functions) is deprecated. Please remove it.

File: ports/libtomcrypt/CONTROL [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Source: libtomcrypt
+Version: 1.18.2
```
Comment (LilyWangL) [inline ports/libtomcrypt/CONTROL]: Can you please add the Homepage for this port in the CONTROL file?
```
Homepage: https://github.com/libtom/libtomcrypt
```

File: ports/libtommath/CONTROL [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Source: libtommath
+Version: 1.2.0
```
Comment (LilyWangL) [inline ports/libtommath/CONTROL]: Can you please add the Homepage for this port in the CONTROL file?
```
Homepage: https://github.com/libtom/libtommath
```

**Review by LilyWangL — CHANGES_REQUESTED** [review]

**Comment by LilyWangL** [comment]
If this port doesn't support some triplets, please so these steps as below:

1. Add [vcpkg_fail_port_install](https://github.com/microsoft/vcpkg/blob/f5c732b40d43f062278f247036b773477823813b/docs/maintainers/vcpkg_fail_port_install.md#vcpkg_fail_port_install) to portfile.cmake.
2. Add the following code to _VCPKG_PATH/scripts/ci.baseline.txt_:
```
libtomcrypt:arm64-windows=fail
libtomcrypt:arm-uwp=fail
libtomcrypt:x64-windows-static=fail
libtommath:x64-osx
libtommath:x64-linux
```

**Comment by FredeJ** [comment]
I would love to just add support for triplets, but I have no idea how to do it - is there any chance that triplet support could be added to the https://github.com/microsoft/vcpkg/blob/master/docs/examples/packaging-github-repos.md guide? The only thing I can find guidance for is how to add a _new_ triplet, not how to use existing.

**Comment by LilyWangL** [comment]
Our CI testing system will test if the port can build successfully on the following triplets, you can click **Details** to check the build error log.
```
x86-windows
x64-windows
x64-windows-static
x64-uwp
x64-osx
x64-linux
arm-uwp
arm64-windows
```
Now the port build failed on some triplets as below. If this result is expected, please do the above steps:
```
libtomcrypt:arm64-windows=fail
libtomcrypt:arm-uwp=fail
libtomcrypt:x64-windows-static=fail
libtommath:x64-osx
libtommath:x64-linux
```

File: ports/libtomcrypt/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    ${SOURCE_PATH}/src/headers/
+    DESTINATION ${CURRENT_PACKAGES_DIR}/include
+)
+
+#Copy license
+file(
+    INSTALL 
+    ${SOURCE_PATH}/LICENSE
+    DESTINATION ${CURRENT_PACKAGES_DIR}/share/libtomcrypt/ RENAME copyright
+)
```
Comment (NancyLi1013) [inline ports/libtomcrypt/portfile.cmake]: Could you please update
```
file(
    INSTALL 
    ${SOURCE_PATH}/LICENSE
    DESTINATION ${CURRENT_PACKAGES_DIR}/share/libtomcrypt/ RENAME copyright
) 
```
as
```
file(
    INSTALL 
    ${SOURCE_PATH}/LICENSE
    DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT}  RENAME copyright
) 
```
?

File: ports/libtommath/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+file(INSTALL
+    ${SOURCE_PATH}/tommath.h
+    DESTINATION ${CURRENT_PACKAGES_DIR}/include
+)
+
+file(INSTALL
+    ${SOURCE_PATH}/LICENSE
+    DESTINATION ${CURRENT_PACKAGES_DIR}/share/libtommath/copyright
+)
```
Comment (NancyLi1013) [inline ports/libtommath/portfile.cmake]: Could you please update 
```
file(INSTALL
    ${SOURCE_PATH}/LICENSE
    DESTINATION ${CURRENT_PACKAGES_DIR}/share/libtommath/copyright
)
```
as
```
file(INSTALL
    ${SOURCE_PATH}/LICENSE
    DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright
)
```
?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @FredeJ 
Could you please address the review suggestions and regressions?

**Comment by LilyWangL** [comment]
Pinging @FredeJ for response. Is work still being done for this PR?

**Comment by LilyWangL** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Pull request closed without merging**